### PR TITLE
Use return types and phpdoc

### PR DESCRIPTION
**Pull request by whikloj**
This is to add some documentation (as I understand it) and types to arguments and functions. 

This should have no impact on the code...except possibly for ACK/NACK. 

It appears that Apollo sends a "ack" header on MESSAGE frames which is not part of the spec. That seems to have impacted the base `Protocol` class which was looking for this `ack` header in the `Protocol::getAckFrame` function. 

Also but unrelated the in `Protocol::getNackFrame` a `message-id` header was being sent on all "nack" messages and was overwriting what may have been set previously.

### CODE REVIEW

**Review by staabm — COMMENTED** [review]
I like adding native types. It will require a new major version I guess, as adding parameter types can be a breaking change for existing code

**Comment by whikloj** [comment]
Hey @staabm,

I am good with making a new major release for this change.  If you have time to review and merge this, we can cut a 6.0.0 release

**Comment by staabm** [comment]
I will have a closer look in the next view days

**Comment by staabm** [comment]
to make sure phpstan reports nullability issues we need either raise the phpstan level (which would also bring in a of other non-nullability related errors) or we enable nullability checking separately

see https://twitter.com/markusstaab/status/1710201565087256893

File: tests/Unit/Protocol/ProtocolTestCase.php [hunk 1]
```diff
@@ -90,11 +90,11 @@ public function testAckVersionTwo()
         $instance = $this->getProtocol(Version::VERSION_1_2);
 
         $actual = $instance->getAckFrame(
-            new Frame(null, ['message-id' => 'id-value', 'ack' => 'ack-id-value']),
```
Comment (staabm) [inline tests/Unit/Protocol/ProtocolTestCase.php:93]: are these test-changes related to the type changes, or could be separated because unrelated?

**Review by staabm — COMMENTED** [review]
~github bug~

**Comment by whikloj** [comment]
What I discovered when starting this is that almost everything allowed `null`, because `null` gets passed around as defaults and in tests.

Adding types caused these `nulls` to cause errors so tests and codestyle and phpstan would not pass.

So I'm going to close this and consider removing all the uses of/acceptance of `null` first.

Then try types again. I will consider the other comments in the other PRs

**Pull request closed without merging**